src/platform/modulesmanager/src/ModulesManager.cpp

```diff
@@ -440,194 +463,139 @@ void ModulesManager::UnloadAllModules()
+                OsConfigLogError(ModulesManagerLog::Get(), "MpiSet(%s, %s, -, %d) returned %d", componentName, objectName, payloadSizeBytes, status);
             }
         }
     }};
 
-    if (nullptr == moduleMetadata)
+    if (nullptr == componentName)
     {
         OsConfigLogError(ModulesManagerLog::Get(), "MpiSet invalid componentName: %s", componentName);
-        retValue = ENOENT;
-        return retValue;
+        status = EINVAL;
     }
-
-    bool timeout = ManagementModule::Lifetime::KeepAlive != moduleMetadata->module->GetLifetime();
-    moduleMetadata->operationInProgress = true;
-
-    retValue = MpiSetInternal(componentName, propertyName, payload, payloadSizeBytes);
-
-    // Schedule an unload module if theres a timeout allowed
-    if (timeout)
+    else if (nullptr == objectName)
     {
-        ScheduleUnloadModule(*moduleMetadata);
+        OsConfigLogError(ModulesManagerLog::Get(), "MpiSet invalid objectName: %s", objectName);
+        status = EINVAL;
     }
-
-    return retValue;
-}
-
-int ModulesManager::MpiSetInternal(const char* componentName, const char* propertyName, const char* payload, const int payloadSizeBytes)
-{
-    // Dispatch call to the right module
-    if (modMap.end() == modMap.find(componentName))
+    else if (nullptr == payload)
     {
-        OsConfigLogError(ModulesManagerLog::Get(), "Unable to find %s in module map", componentName);
-        return EINVAL;
+        OsConfigLogError(ModulesManagerLog::Get(), "MpiSet invalid payload");
+        status = EINVAL;
     }
-    ModuleMetadata &moduleMetadata = modMap[componentName];
-    moduleMetadata.lastOperation = std::chrono::system_clock::now();
-    int ret = moduleMetadata.module->MmiSet(componentName, propertyName, (MMI_JSON_STRING)payload, payloadSizeBytes);
-    moduleMetadata.operationInProgress = false;
-
-    if (MMI_OK == ret)
+    else if (0 >= payloadSizeBytes)
     {
-        if (IsFullLoggingEnabled())
-        {
-            OsConfigLogInfo(ModulesManagerLog::Get(), "MmiSet(%s, %s, %.*s, %d) to %s returned %d", componentName, propertyName, payloadSizeBytes, payload, payloadSizeBytes, moduleMetadata.module.get()->GetName().c_str(), ret);
-        }
-        else
-        {
-            OsConfigLogInfo(ModulesManagerLog::Get(), "MmiSet(%s, %s, -, %d) to %s returned %d", componentName, propertyName, payloadSizeBytes, moduleMetadata.module.get()->GetName().c_str(), ret);
-        }
+        OsConfigLogError(ModulesManagerLog::Get(), "MpiSet invalid payloadSizeBytes: %d", payloadSizeBytes);
+        status = EINVAL;
     }
     else
     {
-        if (IsFullLoggingEnabled())
-        {
-            OsConfigLogError(ModulesManagerLog::Get(), "MmiSet(%s, %s, %.*s, %d) to %s returned %d", componentName, propertyName, payloadSizeBytes, payload, payloadSizeBytes, moduleMetadata.module.get()->GetName().c_str(), ret);
-        }
-        else
+        ModuleMetadata& moduleMetadata = modMap[componentName];

```

Comment:
>modMap[componentName]; [](http://example.com/codeflow?start=41&length=22)

This looks unsafe. componentName can be anything

Reply:
Agreed, I will harden this